### PR TITLE
chore: simplify integration tests

### PR DESCRIPTION
**Pull request by igorbernstein2**
* switch all integration tests to use failsafe-plugin
* remove api-gap-test execution as it wasnt used
* stop override report suffixes so that sponge suffixes are not clobbered (test results will be visible in test fusion)
* use a separate directory for each test execution to avoid clobbering results
* remove all suite subclasses and configure category filtering into plugin config (eventually suites will go away altogether)

### CODE REVIEW

**Review by kolea2 — COMMENTED** [review]
overall looks good, a few questions in the comment and also:

Can you please confirm the ITs currently look correct (ran against prod and not the emulator or minicluster)?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 1]
```diff
@@ -259,31 +142,24 @@ limitations under the License.
                 </goals>
                 <configuration>
                   <includes>
-                    <include>**/EmulatorTests.java</include>
+                    <include>**/IntegrationTests.java</include>
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml:145]: not sure what you mean. Can you rephrase?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 2]
```diff
@@ -45,180 +44,64 @@ limitations under the License.
...
-                  <!-- to enable netty logging, include:
-                  -Djava.util.logging.config.file=src/test/resources/logging.properties
-                  -->
                   <forkCount>1</forkCount>
                   <includes>
-                    <include>**/IntegrationTestsGap.java</include>
+                    <include>**/IntegrationTests.java</include>
                   </includes>
-                  <reportNameSuffix>bigtable-server</reportNameSuffix>
+                  <excludedGroups>KnownGap</excludedGroups>
```
[the remark this answers is not in the document]
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml:60]: We arent, but they exist and I didnt want to delete them just yet. So for know I'm retaining the old behavior of excluding them

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kolea2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml:145]: Why do we have this entire section? What is it running?

[the remark this answers is not in the document]
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml:145]: all of the integrations tests against the emulator

[the remark this answers is not in the document]
Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml:145]: the plugin above will start an emulator and will inject the emulator endpoint into this execution and run all of the tests that arent tagged with KnownEmulatorGap or KnownGap

**Review by kolea2 — APPROVED** [review]
